addons/popochiu/engine/objects/room/popochiu_room_data.gd

```diff
@@ -120,13 +120,24 @@ func save_children_states() -> void:
 						continue
 					# ---------------------------------------------------------------- Fix #320 ----
 					
-					var script_path := scene_path.replace("tscn", "gd")
-					if not FileAccess.file_exists(script_path):
+					var packed_scene: PackedScene = load(scene_path)
+					if not packed_scene or not packed_scene is PackedScene:
 						folder_name = dir.get_next()
 						continue
 					
-					var node: Node2D = load(script_path).new()
-					node.script_name = folder_name.to_pascal_case()
+					# Instantiate the full scene so exported data (like walkable polygons) is already baked.
+					# Loading only the script left interaction_polygon empty, so the first time we entered
+					# a room the stored state overwrote its walkable areas with blank data. Grabbing the
+					# packed scene preserves those serialized outlines until the room can save itself.
+					var node: Node = packed_scene.instantiate()
```

Comment:
I get the point of doing this, but there’s something I keep thinking about: as far as I remember, the baked info for navigation and interaction polygons is saved in the Room, not in each WalkableArea or each Prop/Hotspot. If this fixes the bug, we can leave it as is, but maybe we should check whether we’re actually handling that info correctly. Maybe the more logical solution would be to save that data in the corresponding WalkableArea or Clickable scene when the Room gets modified.